SignalServiceKit/src/Util/OWSSwiftUtils.swift

```diff
@@ -20,3 +20,16 @@ public func owsFail(_ message: String) {
     Logger.flush()
     assertionFailure(message)
 }
+
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+@objc
+public func AssertIsOnMainThread(file: String = #file,
```

Comment:
substantive change 2: add this swift version. (all other changes are either a straight search/replace auto-formatting)

We can leverage the `#file`/`#function`/`#line` as default attributes to make a cleaner call site.

(i.e. `AssertIsOnMainThread()` vs `AssertIsOnMainThread(#function)`)

I'm considering doing something similar for `owsFail` and `Logger.foo` methods so we can stop doing the `"\(logTag) in \(#function)"` dance, but that would require a slightly more hands on approach. WDYT @charlesmchen?

Reply:
Noice

> I'm considering doing something similar for owsFail and Logger.foo methods so we can stop doing the "\(logTag) in \(#function)" dance, but that would require a slightly more hands on approach. WDYT @charlesmchen?

I support this.  

I assume `#file/#function/#line` work in free functions and don't capture a reference to self if used in a block.

Reply:
Can this method be called from obj-c code?  If so, I suggest we call it `SwiftAssertIsOnMainThread` to help visually distinguish it from `OWSAssertIsOnMainThread()`.

Reply:
I removed the `@objc`. That was a mistake, I only intended it for Swift.